### ISSUE TITLE
Inconsistent quad interface compared to Gurobi.jl

### ISSUE THREAD

**Issue by sbebo** [JuliaOpt/CPLEX.jl#37]
The methods to build a quadratic objective function are somewhat confusing, it is not clear when one must pass the matrix coefficients or the "term" coefficients.
Even more confusing when you consider that the convention used in CPLEX.jl is different from what is used in Gurobi.jl: `add_qpterms()` expects "term" coefficients in Gurobi.jl, while it expects "matrix" coefficients in CPLEX.jl. 
This is probably the cause why the test qp_01.jl gives "wrong" results in CPLEX.jl compared to what is expected (the results are not  actually checked in the tests) -- it was probably copied & pasted from GUROBI.jl, that uses a different convention. 

Not saying that the two interfaces have to be identical -- they don't, but I found it somewhat confusing. 

[Also, the MPB tests do not pass, `updatemodel()` doesn't exist for CPLEX.]

**Comment by joehuchette** [comment]
I don't have a strong preference; if I remember correctly, we chose to mirror closely how CPLEX wanted the data at this level, and then used a middle-level fallback in MathProgBase to convert the "terms" form to the "matrix" form.

Agreed that the interface is confusing; that's because this was developed as an "internal" library for JuMP (meaning that the user should never have to touch the CPLEX.jl functions/objects themselves). Perhaps it's time to clean this up and standardize the interface. PRs or suggestions welcome.

Regarding ``updatemodel``, I think we've moved away from forcing solvers to implement methods that aren't necessary like this, so the MPB tests should be changed to reflect this.
